### PR TITLE
[Feat] Add more color to types file

### PR DESCRIPTION
**Pull request by cuzik**
Hi,

I'm proposing an improve of the standard colors.

- [x] Add more basic colors (shade of gray)
- [x] Add more extra colors

### CODE REVIEW

**Comment by stephanemagnenat** [comment]
Thank you for your PR!

I'll be happy to merge it, following two small changes:
- Please remove the whitespace-only changes.
- When a color is composed of two words (such as "light blue"), blue use the camelCase convention used elsewhere in Enki; (example `lightBlue`).

**Comment by cuzik** [comment]
Hi, I will gladly make the changes

File: enki/Types.cpp [hunk 1]
```diff
@@ -47,7 +47,7 @@ namespace Enki
 		const unsigned b((color>>0)&0xff);
 		return Color(double(r)/255., double(g)/255., double(b)/255., double(a)/255.);
 	}
-	
+
```
Comment (cuzik) [inline enki/Types.cpp:50]:
```suggestion
	
```

File: enki/Types.cpp [hunk 2]
```diff
@@ -7,8 +7,8 @@
     Copyright (C) 2006-2008 Laboratory of Robotics Systems, EPFL, Lausanne
     See AUTHORS for details
 
-    This program is free software; the authors of any publication 
-    arising from research using this software are asked to add the 
+    This program is free software; the authors of any publication
+    arising from research using this software are asked to add the
```
Comment (cuzik) [inline enki/Types.cpp:11]:
```suggestion
    This program is free software; the authors of any publication 
    arising from research using this software are asked to add the 
```

File: enki/Types.h [hunk 3]
```diff
@@ -7,8 +7,8 @@
     Copyright (C) 2006-2008 Laboratory of Robotics Systems, EPFL, Lausanne
     See AUTHORS for details
 
-    This program is free software; the authors of any publication 
-    arising from research using this software are asked to add the 
+    This program is free software; the authors of any publication
+    arising from research using this software are asked to add the
```
Comment (cuzik) [inline enki/Types.h:11]:
```suggestion
    This program is free software; the authors of any publication 
    arising from research using this software are asked to add the 
```

File: enki/Types.h [hunk 4]
```diff
@@ -56,7 +56,7 @@ namespace Enki
 	{
 		//! RGBA values in range [0..1]
 		double components[4];
-		
+
```
Comment (cuzik) [inline enki/Types.h:59]:
```suggestion
		
```

File: enki/Types.h [hunk 5]
```diff
@@ -65,44 +65,44 @@ namespace Enki
 			components[2] = b;
 			components[3] = a;
 		}
-		
+
```
Comment (cuzik) [inline enki/Types.h:68]:
```suggestion
		
```

File: enki/Types.h [hunk 6]
```diff
@@ -65,44 +65,44 @@ namespace Enki
...
 		//! access component i
 		const double& operator[](size_t i) const { assert(i < 4); return components[i]; }
 		//! access component i
 		double& operator[](size_t i) { assert(i < 4); return components[i]; }
-		
+
```
Comment (cuzik) [inline enki/Types.h:73]:
```suggestion
		
```

File: enki/Types.h [hunk 7]
```diff
@@ -65,44 +65,44 @@ namespace Enki
...
 		// operations with scalar
 		//! Add d to each component
 		void operator +=(double d) { for (size_t i=0; i<3; i++) components[i] += d; }
 		//! Add d to each component and return result in a new color. I'm left unchanged
 		Color operator +(double d) const { Color c; for (size_t i=0; i<3; i++) c.components[i] = components[i] + d; return c; }
-		
+
```
Comment (cuzik) [inline enki/Types.h:79]:
```suggestion
		
```

File: enki/Types.h [hunk 8]
```diff
@@ -65,44 +65,44 @@ namespace Enki
...
 		//! Substract d from each component
 		void operator -=(double d) { for (size_t i=0; i<3; i++) components[i] -= d; }
 		//! Substract d from each component and return result in a new color. I'm left unchanged
 		Color operator -(double d) const { Color c; for (size_t i=0; i<3; i++) c.components[i] = components[i] - d; return c; }
-		
+
```
Comment (cuzik) [inline enki/Types.h:84]:
```suggestion
		
```

File: enki/Types.h [hunk 9]
```diff
@@ -65,44 +65,44 @@ namespace Enki
...
 		//! Multiply each component with d
 		void operator *=(double d) { for (size_t i=0; i<3; i++) components[i] *= d; }
 		//! Multiply each component with d and return result in a new color. I'm left unchanged
 		Color operator *(double d) const { Color c; for (size_t i=0; i<3; i++) c.components[i] = components[i] * d; return c; }
-		
+
```
Comment (cuzik) [inline enki/Types.h:89]:
```suggestion
		
```

File: enki/Types.h [hunk 10]
```diff
@@ -65,44 +65,44 @@ namespace Enki
...
 		//! Divide each component with d
 		void operator /=(double d) { for (size_t i=0; i<3; i++) components[i] /= d; }
 		//! Divide each component with d and return result in a new color. I'm left unchanged
 		Color operator /(double d) const { Color c; for (size_t i=0; i<3; i++) c.components[i] = components[i] / d; return c; }
-		
+
```
Comment (cuzik) [inline enki/Types.h:94]:
```suggestion
		
```

File: enki/Types.h [hunk 11]
```diff
@@ -65,44 +65,44 @@ namespace Enki
...
 		// operation with another color
 		//! Add oc's components to ours
 		void operator +=(const Color &oc) { for (size_t i=0; i<3; i++) components[i] += oc.components[i]; }
 		//! Add oc's components to ours and return result in a new color. I'm left unchanged
 		Color operator +(const Color &oc) const { Color c; for (size_t i=0; i<3; i++) c.components[i] = components[i] + oc.components[i]; return c; }
-		
+
```
Comment (cuzik) [inline enki/Types.h:100]:
```suggestion
		
```

File: enki/Types.h [hunk 12]
```diff
@@ -65,44 +65,44 @@ namespace Enki
...
 		//! Substract oc's components to ours
 		void operator -=(const Color &oc) { for (size_t i=0; i<3; i++) components[i] -= oc.components[i]; }
 		//! Substract oc's components to ours and return result in a new color. I'm left unchanged
 		Color operator -(const Color &oc) const { Color c; for (size_t i=0; i<3; i++) c.components[i] = components[i] - oc.components[i]; return c; }
-		
+
```
Comment (cuzik) [inline enki/Types.h:105]:
```suggestion
		
```

File: enki/Types.h [hunk 13]
```diff
@@ -111,60 +111,92 @@ namespace Enki
 		void threshold(const Color &limit) { for (size_t i=0; i<3; i++) components[i] = components[i] > limit.components[i] ? components[i] : 0; }
 		//! Return the grey level value
 		double toGray() const { return (components[0] + components[1] + components[2]) / 3; }
-		
+
```
Comment (cuzik) [inline enki/Types.h:114]:
```suggestion
		
```

File: enki/Types.h [hunk 14]
```diff
@@ -111,60 +111,92 @@ namespace Enki
...
 		//! Return a string describing this color
 		std::string toString() const { std::ostringstream oss; oss << *this; return oss.str(); }
-		
+
```
Comment (cuzik) [inline enki/Types.h:117]:
```suggestion
		
```

File: enki/Types.h [hunk 15]
```diff
@@ -111,60 +111,92 @@ namespace Enki
...
 		//! Red component value getter
 		double r() const { return components[0]; }
-		
+
```
Comment (cuzik) [inline enki/Types.h:120]:
```suggestion
		
```

File: enki/Types.h [hunk 16]
```diff
@@ -111,60 +111,92 @@ namespace Enki
...
 		//! Set the value of red component
 		void setR(double value) { components[0] = value; }
-		
+
```
Comment (cuzik) [inline enki/Types.h:123]:
```suggestion
		
```

File: enki/Types.h [hunk 17]
```diff
@@ -111,60 +111,92 @@ namespace Enki
...
 		//! Green component value getter
 		double g() const { return components[1]; }
-		
+
```
Comment (cuzik) [inline enki/Types.h:126]:
```suggestion
		
```

File: enki/Types.h [hunk 18]
```diff
@@ -111,60 +111,92 @@ namespace Enki
...
 		//! Set the value of green component
 		void setG(double value) { components[1] = value; }
-		
+
```
Comment (cuzik) [inline enki/Types.h:129]:
```suggestion
		
```

File: enki/Types.h [hunk 19]
```diff
@@ -111,60 +111,92 @@ namespace Enki
...
 		//! Blue component value getter
 		double b() const { return components[2]; }
-		
+
```
Comment (cuzik) [inline enki/Types.h:132]:
```suggestion
		
```

File: enki/Types.h [hunk 20]
```diff
@@ -111,60 +111,92 @@ namespace Enki
...
 		//! Set the value of blue component
 		void setB(double value) { components[2] = value; }
-		
+
```
Comment (cuzik) [inline enki/Types.h:135]:
```suggestion
		
```

File: enki/Types.h [hunk 21]
```diff
@@ -111,60 +111,92 @@ namespace Enki
...
 		//! Alpha component value getter
 		double a() const { return components[3]; }
-		
+
```
Comment (cuzik) [inline enki/Types.h:138]:
```suggestion
		
```

File: enki/Types.h [hunk 22]
```diff
@@ -111,60 +111,92 @@ namespace Enki
...
 		//! Set the value of alpha component
 		void setA(double value) { components[3] = value; }
-		
+
```
Comment (cuzik) [inline enki/Types.h:141]:
```suggestion
		
```

File: enki/Types.h [hunk 23]
```diff
@@ -111,60 +111,92 @@ namespace Enki
...
 		//! Build from an ARGB uint32_t (0xAARRGGBB in little endian)
 		static Color fromARGB(uint32_t color);
 		//! Build from an ABGR uint32_t (0xAABBGGRR in little endian)
 		static Color fromABGR(uint32_t color);
 		//! Pack into ABGR uint32_t (0xAABBGGRR in little endian)
 		static uint32_t toARGB(Color color);
-		
+
```
Comment (cuzik) [inline enki/Types.h:148]:
```suggestion
		
```

File: enki/Types.h [hunk 24]
```diff
@@ -111,60 +111,92 @@ namespace Enki
...
+		static const Color purple;
+		//! pink (1, 0, 0.8)
+		static const Color pink;
+		//! cyan (0, 1, 1)
+		static const Color cyan;
+
 		friend std::ostream & operator<<(std::ostream &os, const Color& c);
 	};
-	
+
```
Comment (cuzik) [inline enki/Types.h:196]:
```suggestion
	
```

File: enki/Types.h [hunk 25]
```diff
@@ -111,60 +111,92 @@ namespace Enki
...
 	//! A texture
 	typedef std::vector<Color> Texture;
-	
+
```
Comment (cuzik) [inline enki/Types.h:199]:
```suggestion
	
```

**Comment by stephanemagnenat** [comment]
Merged, Thank you!

**Pull request merged**